message/validation/genesis/consensus_validation.go

```diff
@@ -131,7 +140,7 @@ func (mv *messageValidator) validateConsensusMessage(
 			signerState.Reset(msgRound)
 		}
 
-		if mv.hasFullData(signedMsg) && signerState.ProposalData == nil {
+		if mv.hasFullData(signedMsg) && signerState.ProposalData == nil && signedMsg.Message.MsgType == genesisspecqbft.ProposalMsgType {
```

Comment:
I'd check `signedMsg.Message.MsgType == genesisspecqbft.ProposalMsgType` first for readability

Reply:
Updated. Thanks!